### PR TITLE
Tests: Adopt do_regtest.py for CMake builds

### PR DESCRIPTION
**Pull request by oschuett**
See also discussion in #3314.
CC: @jpoto.

### CODE REVIEW

**Comment by jpoto** [comment]
I tried your branch and the testing works if:
1) you give the full path to the executable (not relative one)
2) Define an install directory and run: make install
(otherwise the data directory is not found)

**Comment by oschuett** [comment]
Thanks for testing @jpoto! It now also works with relative paths.

As for the data dir, I guess it should depend on `-DCMAKE_BUILD_TYPE`. For `Debug` it should point to the data dir in the source tree and for `Release` to the install location.

@LecrisUT, I also tried to adopt your Fedora files, but I wasn't quite sure what to do with [do_regtest.sh](https://github.com/cp2k/cp2k/blob/master/tools/fedora/tests/do_regtest.sh). Could you please take a look?

File: tools/fedora/cp2k.spec [hunk 1]
```diff
@@ -250,7 +247,7 @@ for mpi in '' mpich %{?with_openmpi:openmpi} ; do
   # Run packaged do_regtest.sh with appropriate buildroot runpaths
   env PATH=%{buildroot}${bindir}:${PATH} \
     LD_LIBRARY_PATH=%{buildroot}${libdir} \
-    tests/do_regtest.py ${test_common_args[@]} ${test_mpi_args[@]}
+    tests/do_regtest.py %{_bindir} ${test_mpi_args[@]}
```
Comment (LecrisUT) [inline tools/fedora/cp2k.spec]:
```suggestion
    tests/do_regtest.py %{buildroot}${bindir} ${test_mpi_args[@]}
```
If you want to point it to "installed" path
```suggestion
    tests/do_regtest.py %{_vpath_builddir}/bin ${test_mpi_args[@]}
```
If you want to point it to the build path

**Review by LecrisUT — CHANGES_REQUESTED** [review]
`do_regtest.sh` runs the tests as if they are installed. Presumably what you want is to change:
https://github.com/cp2k/cp2k/blob/6eaca959c02db393e2626197232d94e616a58de8/tools/fedora/tests/do_regtest.sh#L25
into 
```diff
-      rlRun "args=\"\$args local ssmp\"" 0 "Set run specific arguments"
+      rlRun "args=\"\$args /usr/bin ssmp\"" 0 "Set run specific arguments"
```
Similarly for the part above

**Comment by LecrisUT** [comment]
Please also check `CP2K_STEM` in `do_regtest.py`. It seems that it becomes redundant

**Pull request merged**